Makefile

```diff
@@ -1,79 +1,185 @@
+
+# JS test is local only
+.PHONY: test-js
+test-js:
+	$(JS_DOCKER_CMD) sh -c 'cd brigade-worker && yarn build && yarn test'
+
+################################################################################
+# Build / Publish                                                              #
+################################################################################
+
+build: build-all-images build-brig
+
+.PHONY: build-all-images
+build-all-images: $(addsuffix -build-image,$(IMAGES))
+
+%-build-image:
+	docker build \
+		-f $*/Dockerfile \
+		-t $(DOCKER_IMAGE_PREFIX)$*:$(IMMUTABLE_DOCKER_TAG) \
+		--build-arg LDFLAGS='$(LDFLAGS)' \
+		.
+	docker tag $(DOCKER_IMAGE_PREFIX)$*:$(IMMUTABLE_DOCKER_TAG) $(DOCKER_IMAGE_PREFIX)$*:$(MUTABLE_DOCKER_TAG)
+
+# Cross-compile binaries for brig
+build-brig:
+	$(GO_DOCKER_CMD) bash -c 'LDFLAGS="$(LDFLAGS)" scripts/build-brig.sh'
+
+.PHONY: push
+push: push-all-images
 
 # You must be logged into DOCKER_REGISTRY before you can push.
-.PHONY: docker-push
-docker-push: $(addsuffix -push,$(IMAGES))
-
-%-push:
-	docker push $(DOCKER_REGISTRY)/$*:$(IMAGE_TAG)
-
-# Cross-compile binaries for our CX targets.
-# Mainly, this is for brig-cross-compile
-%-cross-compile: vendor
-	@for os in $(CX_OSES); do \
-		echo "building $$os"; \
-		for arch in $(CX_ARCHS); do \
-			GOOS=$$os GOARCH=$$arch CGO_ENABLED=0 go build -ldflags '$(LDFLAGS)' -o ./bin/$*-$$os-$$arch ./$*/cmd/$*; \
-		done; \
-		if [ $$os = 'windows' ]; then \
-			mv ./bin/$*-$$os-$$arch ./bin/$*-$$os-$$arch.exe; \
-		fi; \
-	done
-
-.PHONY: build-release
-build-release: brig-cross-compile
+.PHONY: push-all-images
+push-all-images: build-all-images
+push-all-images: $(addsuffix -push-image,$(IMAGES))
+
+%-push-image:
+	docker push $(DOCKER_IMAGE_PREFIX)$*:$(IMMUTABLE_DOCKER_TAG)
+	docker push $(DOCKER_IMAGE_PREFIX)$*:$(MUTABLE_DOCKER_TAG)
+
+################################################################################
+# Helm and functional test utils                                               #
+################################################################################
+
+# Helm chart/release defaults
+BRIGADE_RELEASE                 ?= brigade-server
+BRIGADE_NAMESPACE               ?= default
+BRIGADE_GITHUB_GW_SERVICE       := $(BRIGADE_RELEASE)-brigade-github-app
+BRIGADE_GITHUB_GW_INTERNAL_PORT := 80
+BRIGADE_GITHUB_GW_EXTERNAL_PORT := 7744
+BRIGADE_VERSION := $(VERSION)
+
+# DOCKER_IMAGE_PREFIX would have a trailing slash that we won't want here
+# because the charts glue the registry (registry + org, really), image name.
+# and tag together themselves and do not account for the possibility of registry
+# or org being "". So, we create a new variable here called BRIGADE_REGISTRY
+# that is DOCKER_IMAGE_PREFIX minus any trailing slash. If DOCKER_IMAGE_PREFIX
+# is the empty string, we default to "brigadecore", since, again, the charts
+# cannot currently accommodate a blank registry or org.
+ifeq ("$(DOCKER_IMAGE_PREFIX)","")
```

Comment:
I'm wondering if we should add a note around needing to set `DOCKER_ORG` to `brigadecore` when running `make build-all-images` such that the images can be pulled when/if moving on to `make helm-install/upgrade test-functional`, etc.?  Or, I suppose, more generally, needing to set `DOCKER_ORG` to _something_...

(My particular use case was running/using the minikube docker daemon but, when no DOCKER_REGISTRY/ORG vals were set in my env, `make build-all-images` produced the bare, non-org'd images, e.g., `brigade-worker:edge`, etc. -- however, the helm release defaults to `brigadecore/brigade-worker:edge`)

Reply:
`checkRun` was blindly copied from the github gw's `brigade.js`. Whoops. This is fixed in a recent commit.

Reply:
> make build-all-images produced the bare, non-org'd images, e.g., brigade-worker:edge, etc.

We had discussed this briefly... this was a deliberate choice to reduce (albeit not eliminate) the possibility of individuals with write permissions to the official Docker repo from accidentally building and pushing images from local source. I stand by this being a good idea...

But, I think you're absolutely right that some doc updates, at least are in order, and that maybe the `helm-upgrade` make target should be updated to honor the same env vars. I'm on it...

Reply:
Or, actually, in your specific case, you can probably skip the push and not even bother at all with setting `DOCKER_ORG` at all since you're building images and and pulling them in the same VM.

Just `make build-all-images helm-upgrade` should work, I think. lmk if I am wrong. It happens frequently.

Definitely need to (and will) document this better.

Reply:
Updated docs in a recent commit.

Reply:
> Just make build-all-images helm-upgrade should work

This doesn't quite work on account of the `brigade` chart concatenating values to construct the full image:tag string (and so, always prefacing an un-org'd image name with a `/`, as you pointed out in comments, I believe.)  However, the added/updated docs give good examples of how to run these targets with `DOCKER_REGISTRY`/`DOCKER_ORG` overrides, so I think we're good.